packages/components/pie-modal/src/index.ts

```diff
@@ -140,32 +144,41 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
     /**
      * Opens the dialog element and disables page scrolling
      */
-    private _handleModalOpened () : void {
-        const modalScrollContainer = this._dialog?.querySelector('.c-modal-scrollContainer');
+    private _handleModalOpened (event: CustomEvent): void {
+        const { targetModal } = event.detail;
 
-        if (modalScrollContainer) {
-            disableBodyScroll(modalScrollContainer);
-        }
+        if (targetModal === this) {
+            const modalScrollContainer = this._dialog?.querySelector('.c-modal-scrollContainer');
 
-        if (this._dialog?.hasAttribute('open') || !this._dialog?.isConnected) {
-            return;
+            if (modalScrollContainer) {
+                disableBodyScroll(modalScrollContainer);
+            }
+
+            if (this._dialog?.hasAttribute('open') || !this._dialog?.isConnected) {
+                return;
+            }
+
+            // The ::backdrop pseudoelement is only shown if the modal is opened via JS
+            this._dialog?.showModal();
         }
-        // The ::backdrop pseudoelement is only shown if the modal is opened via JS
-        this._dialog?.showModal();
     }
 
     /**
      * Closes the dialog element and re-enables page scrolling
      */
-    private _handleModalClosed () : void {
-        const modalScrollContainer = this._dialog?.querySelector('.c-modal-scrollContainer');
+    private _handleModalClosed (event: CustomEvent): void {
+        const { targetModal } = event.detail;
```

Comment:
this is fantastic @kevinrodrigues - really awesome work!

Reply:
❤️